src/formatDate.js

```diff
@@ -8,7 +8,52 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const partsDate = date.split(fromFormat[3]);
```

Comment:
The use of `fromFormat[3]` to split the date assumes that the separator is always at index 3. This might not be correct if the format array has a different length. Consider dynamically determining the separator based on the format array.